### PR TITLE
feat(esa): add esa provider

### PR DESCRIPTION
**Pull request by suin**
## Summary

[esa](https://esa.io/) is a document-sharing service that helps teams collaboratively grow work-in-progress knowledge into organized documentation.

- add an open-source esa provider with 30 curated actions for teams, posts, comments, categories, attachments, and official help
- support both OAuth 2.0 and esa PAT v2 connections through shared bearer executors
- add bounded response shaping, attachment transit handling, credential validation, and provider-local tests

## Authentication and permissions

OAuth requests esa's provider-native `read` and `write` scopes. PAT v2 permissions are resource-specific; connection validation requires `read:user`, and each enabled action requires its corresponding PAT v2 scope.

OAuth may reach every esa team available to the authorized user. Users who need a team-level boundary should use PAT v2 and configure the token and API access policy in esa.

## Verification

- `npm ci`
- `npm run generate:catalog`
- `npm run lint`
- `npm run typecheck`
- `npm test -- src/providers/esa/definition.test.ts src/providers/esa/executors.test.ts src/providers/esa/runtime.test.ts` — 38 passed
- `npm test` — 824 passed across 77 files

Live verification used a PAT v2 connection restricted to one test team. Team listing returned only that team, and a direct request to another team returned 403. Read actions succeeded, and the create/update/get/delete comment lifecycle completed successfully with the test comment removed afterward.

The three official-help actions were not exercised live because they intentionally access esa's `docs` team; their request and response behavior is covered by automated tests.

### CODE REVIEW

**Comment by suin** [comment]
Working on the fixes. I’ll follow up once they’re pushed.

File: src/providers/esa/runtime.ts [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+  "image/jpeg": true,
+  "image/png": true,
+  "image/gif": true,
+  "image/webp": true,
+};
+const signedAttachmentHosts: Record<string, true> = {
+  "files.esa.io": true,
+  "dl.esa.io": true,
+};
+const publicAttachmentHosts: Record<string, true> = { "img.esa.io": true };
```
[the remark this answers is not in the document]
Comment (suin) [reply src/providers/esa/runtime.ts]: Fixed in f10cc0c. The host and MIME allowlists now use `Set.has()`, with regression coverage for inherited property names.

File: src/providers/esa/runtime.ts [hunk 2]
```diff
@@ -0,0 +1,957 @@
...
+}
+
+async function archivePost(input: Record<string, unknown>, context: EsaActionContext) {
+  const teamName = readTeamName(input);
+  const postNumber = readPositiveInteger(input, "postNumber");
+  const post = await getRawPost(context, teamName, postNumber);
+  const category = optionalRawString(post.category) ?? "";
+  if (category.startsWith("Archived/")) {
+    return { message: "Post is already archived", category };
+  }
```
[the remark this answers is not in the document]
Comment (suin) [reply src/providers/esa/runtime.ts:339]: Fixed in f10cc0c. Top-level `Archived` is now treated as already archived, with both forms covered by the idempotency test.

File: src/providers/esa/runtime.ts [hunk 3]
```diff
@@ -0,0 +1,957 @@
...
+
+function transformBody(bodyMd: string | undefined, options: BodyOptions): string | undefined {
+  if (options.omitBody || bodyMd === undefined) {
+    return undefined;
+  }
+  if (options.truncateBody !== undefined && bodyMd.length > options.truncateBody) {
+    return `${bodyMd.slice(0, options.truncateBody)}\n\n... (truncated)`;
+  }
+  return bodyMd;
+}
```
[the remark this answers is not in the document]
Comment (suin) [reply src/providers/esa/runtime.ts]: Fixed in f10cc0c using the existing `Intl.Segmenter`, so truncation preserves grapheme boundaries and matches `body_md_stats.characters`. Regression coverage includes a multi-code-point emoji.

File: src/providers/esa/runtime.ts [hunk 4]
```diff
@@ -0,0 +1,957 @@
...
+    throw invalidInput("url must use HTTPS");
+  }
+  if (signedAttachmentHosts[url.hostname]) {
+    return { needsSigning: true, path: url.pathname };
+  }
+  if (publicAttachmentHosts[url.hostname]) {
+    return { needsSigning: false, url: url.toString() };
+  }
+  throw invalidInput("url must use img.esa.io, files.esa.io, dl.esa.io, or an /uploads/... path");
+}
```
[the remark this answers is not in the document]
Comment (suin) [reply src/providers/esa/runtime.ts:874]: Fixed in f10cc0c. Attachment downloads now use `providerFetch`, preserving the existing bounded response handling while applying the shared outbound-request protections.

**Comment by Cheerego7** [comment]
> Working on the fixes. I’ll follow up once they’re pushed.

I've reverted my changes, so you can continue addressing the review feedback when you have time.

**Comment by suin** [comment]
Thanks! I’ve pushed the review fixes and incorporated your attachment handling improvements and regression tests in the follow-up commit, with co-authorship credited.

Verified:

- esa provider tests: 53 passed
- full suite: 77 test files, 839 tests passed
- lint, formatting, and typecheck passed
- read-only integration checks confirmed default truncation, full-body retrieval with `truncate: false`, preservation of the original body statistics, and public attachment URL handling

The grapheme-boundary case is covered by automated regression tests.

**Review by Cheerego7 — APPROVED** [review]
